tiledb/common/Cargo.toml

```diff
@@ -0,0 +1,27 @@
+[package]
+name = "tiledb-common"
+edition.workspace = true
+rust-version.workspace = true
+version.workspace = true
+
+[dependencies]
+anyhow = { workspace = true }
+arrow-schema = { workspace = true, optional = true }
+paste = { workspace = true }
+proptest = { workspace = true, optional = true }
+serde = { workspace = true, optional = true }
+serde_json = { workspace = true, optional = true }
+thiserror = { workspace = true }
+tiledb-proc-macro = { workspace = true, optional = true }
+tiledb-sys = { workspace = true }
```

Comment:
I'm reading this PR pretty quickly, so I may have missed an actual ffi call, but I would only expect (and have only seen) references to various constants. We could pretty easily split those off into a tiledb-sys-defs crate to avoid this crate from requiring a link to libtiledb.so. Not needed for this PR, just a thought I had keeping in mind part of the motivation for this PR.

Reply:
Each time I add a new crate I have to add a new `build.rs` file for that crate.  Isn't that what sets the linker path, not the presence of the `sys` crate?

That said I do think it might be nice to split out the typedefs and constants for sys

Reply:
There's two halves: The dependency on tiledb-sys introduces the requirement that libtiledb.so is linked to the final executable. The build.rs in non-tiledb-sys crates tells the linker where to find it on the current system (assuming its not in a system standard path).

So basically, if we split out tiledb-sys-defs that contains just the constant definitions, then anything that doesn't depend on tiledb-sys (directly or indirectly) will no longer need to provide the build.rs.

Or in other words, the build.rs requirement is a symptom of what's actually linking to libtiledb.so, when we remove that need we should be able to remove a lot of the build.rs scripts from the crates that no longer need it.

Reply:
Ok I see that more clearly now from looking at `sys/build.rs`.  Definitely will follow up with that in a subsequent PR.  Thanks!